src/debugger/appWorker.ts

```diff
@@ -186,8 +189,12 @@ export class MultipleLifetimesAppWorker {
         socketToApp.on("message",
             (message: any) => this.onMessage(message));
         socketToApp.on("error",
-            (error: Error) => printDebuggingError("An error ocurred while using the socket to communicate with the React Native app", error));
-        return socketToApp;
+            (error: Error) => {
+                deferred.reject(error);
+            });
+
+        Q.delay(300).done(() => deferred.resolve(socketToApp));
```

Comment:
This would only work if the error happens during the first 300 milliseconds. It may catch a few cases, but I don't think this is a good solution for this issue.